angular/src/lib/search-input/search-input.component.ts

```diff
@@ -0,0 +1,140 @@
+import { Component, OnInit, Input, Output, EventEmitter, Optional, Self} from '@angular/core';
+import { NG_VALUE_ACCESSOR, ControlValueAccessor, NgControl } from '@angular/forms';
+
+const cb = () => {};
+
+@Component({
+  selector: 'cui-search-input',
+  templateUrl: './search-input.component.html',
+  styles: []
+})
+
+export class SearchInputComponent implements ControlValueAccessor {
+
+  /** @prop Optional css class string | ''  */
+  @Input() public class: string = '';
```

Comment:
@pauljeter any idea about how to support ngClass for such components where the classes are applied inside the template

Reply:
No, for classes on internal components, we will expose a prop that we will pass down. i.e. the `inputClass` prop on line 23